src/biogeochem/Fan2CTSMMod.F90

```diff
@@ -0,0 +1,1059 @@
+         
+      end select
+      
+      end associate
+
+    end function get_total_n
+    
+    subroutine balance_check(label, total_old, total_new, flux)
+      ! Check and report that the net flux equals the accumulated mass in pools. The
+      ! total pools and fluxes can be evaluated by the function get_total_n.
+      character(len=*), intent(in) :: label
+      real(r8), intent(in) :: total_old, total_new, flux
+
+      real(r8) :: diff, accflux
+      real(r8) :: tol = 1e-6_r8
+      
+      diff = total_new - total_old
+      accflux = flux*dt
+      
+    end subroutine balance_check
+    
+  end subroutine fan_eval
+
+  !************************************************************************************
+  
+  subroutine handle_storage(bounds, temperature_inst, frictionvel_inst, dt,  &
+       ndep_sgrz_grc, ndep_ngrz_grc, &
+       n_manure_spread_col, tan_manure_spread_col, &
+       n_manure_graze_col, n_manure_mixed_col, &
+       nh3_flux_stores, nh3_flux_barns, man_n_transf, &
+       grz_fract, man_n_barns, tan_fract_excr, &
+       filter_soilc, num_soilc)
+    !
+    ! Evaluate storage losse for manure in the mixed/landless production systems
+    ! associated with crop columns in CLM. The N remaining after storage losses is applied
+    ! on soil either in the same column, or a fraction may be moved to the native
+    ! vegetation column within the gridcell (grasslands). This subroutine also evaluates
+    ! seasonal grazing of livestock, and the manure N on pastures is also moved into the
+    ! native vegatation.
+    ! 
+    use landunit_varcon, only : max_lunit
+    use pftconMod, only : nc4_grass, nc3_nonarctic_grass, nc3_arctic_grass
+    use clm_varcon, only : ispval
+    use landunit_varcon,      only:  istsoil, istcrop
+    use abortutils     , only : endrun
+    use LandunitType   , only: lun
+    use GridcellType   , only: grc
+    use clm_varctl     , only : iulog
+
+    implicit none
+    type(bounds_type), intent(in)    :: bounds
+    type(temperature_type) , intent(in) :: temperature_inst
+    type(frictionvel_type) , intent(in) :: frictionvel_inst
+    real(r8), intent(in) :: dt
+    
+    ! N excreted in manure, gN/m2:
+    real(r8), intent(in) :: ndep_sgrz_grc(bounds%begg:bounds%endg) ! seasonally grazing animals
+    real(r8), intent(in) :: ndep_ngrz_grc(bounds%begg:bounds%endg) ! non-grazing animals
+    ! N, TAN spread on grasslands, gN/m2/s:
+    real(r8), intent(inout) :: n_manure_spread_col(bounds%begc:bounds%endc) 
+    real(r8), intent(out) :: tan_manure_spread_col(bounds%begc:bounds%endc) ! output, calculated from the above and stored manure
+    ! N excreted by animals allocated to mixed production systems temporarily grazing on grasslands:
+    real(r8), intent(inout) :: n_manure_graze_col(bounds%begc:bounds%endc)
+    ! N excreted by animals in mixed systems, total
+    real(r8), intent(out) :: n_manure_mixed_col(bounds%begc:bounds%endc)
```

Comment:
One more thing I noticed: Here and elsewhere, these arrays should be intent(inout) rather than intent(out): They are only set for a subset of points, but making them intent(out) makes their value undefined over other columns. So, for example, if the intent is to have them spval or 0 over columns not set here, the value will actually be undefined upon exit from this subroutine. (Some compilers may maintain the spval or 0 value, but others may not.)

Reply:
Thanks, this was very good comment.